packages/api/src/User/User.service.ts

```diff
@@ -36,4 +38,74 @@ export class UserService {
     user.profileImage = `https://www.gravatar.com/avatar/${md5(input.email)}`;
     return this.userRepository.save(user);
   }
+
+
+  /**
+   * Did the user complete a module yesterday (normalize to 12am)
+   * Returns:
+   *  true = Skipped
+   *  false = Not skippped. Increment
+   *  null = Not skipped. Already completed today. Don't increment
+   * @param userId User to check
+   */
+  private async _skippedDay(userId: string): Promise<boolean | null> {
+    // Normalize to 12am today
+    const now = new Date();
+
+    now.setHours(0);
+    now.setMinutes(0);
+
+    const lastCompleted = await this.userModuleService.findLatestByUser(userId);
+    if (!lastCompleted) return true;
+
+    // Normalize to 12am completed
+    lastCompleted.setHours(0);
+    lastCompleted.setMinutes(0);
+
+    const diffInHours = Math.floor((now.valueOf() - lastCompleted.valueOf()) / 1000 / 60 / 60);
+
+    if (diffInHours === 0) return null; // DON'T increment streak
+
+    return diffInHours > 24; // Did they complete a module yesterday (normalized)
+  }
+
+  /**
+   * retreive and or reset streak of not completeing modules
+   * @param user
+   */
+  async getOrResetStreak(user: UserWithPassword) {
+    const skipped = await this._skippedDay(user.id);
+
+    // If last module was over a day ago, reset back to zero (lost the streak)
+    if (skipped) {
+      await this.userRepository.update({ id: user.id }, { streak: 0 });
```

Comment:
This logic is duplicated in `incrementStreak`, either:

a. Move it to a reusable function
b. Double check if it's really needed in both places

Reply:
Similar to the above, the methods have been updated and their logic streamlined.